src/edu/gatech/cs2340/trydent/TrydentEngine.java

```diff
@@ -0,0 +1,145 @@
+package edu.gatech.cs2340.trydent;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Main class that kicks off the TrydentEngine.
+ * 
+ * As a developer using the TrydentEngine, all you have to do is call
+ * TrydentEngine.start() to get started.
+ * 
+ * @author Garrett Malmquist
+ * */
+public class TrydentEngine {
+
+	private static TrydentEngine instance;
+
+	private Thread mainThread;
+	private final Object mainLock = new Object();
+	
+	private volatile boolean doQuit = false;
+
+	private Set<ContinuousEvent> continuousEvents, continuousEventsToAdd, continuousEventsToRemove;
+	
+
+	private TrydentEngine() {
+		continuousEvents = new HashSet<>();
+		continuousEventsToAdd = new HashSet<>();
+		continuousEventsToRemove = new HashSet<>();
+	}
+
+	private void mainLoop() {
+		Time.startTheDawnOfTime();
+		
+		while (!doQuit) {
+			updateContinuousEvents();
+			Time.updateTime();
+		}
+		
+		cleanup();
+	}
+
+	private void updateContinuousEvents() {
+		continuousEvents.addAll(continuousEventsToAdd);
+		continuousEventsToAdd.clear();
+		
+		// Removing events is uglier than creating events, because
+		// we need to make sure we handle the case where users 
+		// remove additional events in an events onStop() method.
+		while (!continuousEventsToRemove.isEmpty()) {
+			ContinuousEvent next = null;
+			for (ContinuousEvent event : continuousEventsToRemove) {
+				next = event;
+				break;
+			}
+			next.doStop();
+			continuousEvents.remove(next);
+			continuousEventsToRemove.remove(next);
+		}
+		
+		for (ContinuousEvent event : continuousEvents) {
+			event.doUpdate();
+		}
+	}
+	
+	private void cleanup() {
+		// Stop any remaining events.
+		ContinuousEvent[] toStop = new ContinuousEvent[continuousEvents.size()];
+		continuousEvents.toArray(toStop);
+		for (ContinuousEvent event : toStop) {
+			event.doStop();
+		}
+		
+		continuousEvents.clear();
+		continuousEventsToRemove.clear();
+		continuousEventsToAdd.clear();
+		
+		synchronized (mainLock) {
+			// We're done here.
+			mainThread = null;
+		}
+	}
+	
+	private static TrydentEngine getInstance() {
```

Comment:
If we want to make this a singleton, we should probably at least provide a mock class because testing is something that 2340 projects should be able to do easily.